src/main/scala/com/microsoft/hyperspace/index/sources/delta/DeltaLakeRelation.scala

```diff
@@ -142,4 +143,95 @@ class DeltaLakeRelation(spark: SparkSession, override val plan: LogicalRelation)
       /* modificationTime */ modificationTime,
       path)
   }
+
+  private def deltaLakeVersionHistory(index: IndexLogEntry): Seq[(Int, Int)] = {
+    val versions =
+      index.derivedDataset.properties.properties
+        .getOrElse(DeltaLakeConstants.DELTA_VERSION_HISTORY_PROPERTY, "")
+
+    // The value is comma separated versions - <index log version>:<delta table version>.
+    // e.g. "1:2,3:5,5:9"
+    versions.split(",").reverseIterator.foldLeft(Seq[(Int, Int)]()) { (list, versionStr) =>
+      val pair = versionStr.split(":")
+      val Array(indexLogVersion, deltaTableVersion) = versionStr.split(":").map(_.toInt)
+      if (list.nonEmpty && list.head._2 == deltaTableVersion) {
```

Comment:
Can't you do this with `filter` instead of `foldLeft` unless I misunderstood something (in this case, can you add some comments?)

Reply:
Used `foldLeft` to skip the same log version described below.. we could do `filter` with `var` of previous entry.
Which one is more preferable?

Reply:
Ah OK, this is fine then. Could you add a block of comment of some examples for `deltaLakeVersionHistory`?